addons/api/addon/serializers/application.js

```diff
@@ -74,18 +75,38 @@ export default class ApplicationSerializer extends RESTSerializer {
       delete json[key];
     }
     // If an attribute has a `for` option, it must match the
-    // record's `type`, else the attribute is excluded
+    // record's `type`, otherwise the attribute is excluded
     // from serialization.
-    //note: This doesn't handle secrets yet, we can add support for them if needed.
-    if (
-      options?.for &&
-      options.for !== recordType &&
-      !options.for.includes(recordType)
-    ) {
-      if (options.isNestedAttribute) {
-        delete json?.attributes?.[key];
-      } else {
-        delete json[key];
+
+    if (options?.for) {
+      // Check the typeof the `for` option to determine if it is a plugin or not
+      const isPlugin =
+        typeOf(options.for) === 'object' && options.for?.type === 'plugin';
+      const isNotPlugin =
```

Comment:
Do you think it's redundant to have both `isPlugin` and `isNotPlugin`? Are there scenarios where `!isPlugin` won't be the same as `isNotPlugin`?

Reply:
hmm yeah, it does seem redundant, I will leave the comments for more clarity and use `!isPlugin`